### PR TITLE
fix #3933: Duplicated keys in PropTypes table 

### LINKED ISSUE #3933: Multiple empty lines in JSDoc causes key error in React when using Info Addon.
### Bug summary

Creating a JSDoc comment for a prop type with multiple empty lines causes an error in the browser development console.

```
warning.js:33 Warning: Encountered two children with the same key, ``. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.
    in td (created by Td)
    in Td (created by PropTable)
    in tr (created by PropTable)
    in tbody (created by PropTable)
    in table (created by Table)
    in Table (created by PropTable)
    in PropTable
    in Unknown (created by Story)
    in div (created by Story)
    in div (created by Story)
    in div (created by Story)
    in div (created by Story)
    in div (created by Story)
    in Story
```

### Steps to reproduce

1. Create a Story which utilizes the Info Addon:
2. Create a JSDoc comment for a prop type which contains more than one empty line:

Reproduction: https://github.com/strothj/storybook-info-addon-docgen-key-error

### Please specify which version of Storybook and optionally any affected addons that you're running

- @storybook/react 4.0.0-alpha.14
- @storybook/addon-something 4.0.0-alpha.14

### Affected platforms 

- _If UI related, please indicate browser, OS, and version_
- _If dependency related, please include relevant version numbers_
- _If developer tooling related, please include the platform information_

### Screenshots / Screencast / Code Snippets (Optional)

image

1. Create a Story which utilizes the Info Addon:
```
import React from "react";
import { storiesOf } from "@storybook/react";
import SomeComponent from "../SomeComponent";

storiesOf("SomeComponent", module)
  .addParameters({ info: { inline: true } })
  .add("error", () => <SomeComponent />);
```

2. Create a JSDoc comment for a prop type which contains more than one empty line:
```
class SomeComponent extends React.Component {
  static propTypes = {
    /**
     * This is a required prop.
     *
     * With a multiple line description.
     *
     * After second line, causes error.
     */
    color: PropTypes.string
  };

  render() {
    return <div>Some Component</div>;
  }
}

export default SomeComponent;
```

### PR DESCRIPTION
**Pull request by gombek**
Issue: #3933 

## What I did
I've added check for duplicated lines, if line of text occur multiple times, index value will be added (as postfix) to the key.

## How to test

Appropriate test is included in this PR.

### CODE REVIEW

File: addons/info/src/components/PropTable.js [hunk 1]
```diff
@@ -19,7 +22,9 @@ export const multiLineText = input => {
         lineOfText,
         i // note: lineOfText is the closest we will get to a unique key
```
Comment (wuweiweiwu) [inline addons/info/src/components/PropTable.js]: Comment is misleading. Please remove

File: addons/info/src/components/PropTable.js [hunk 2]
```diff
@@ -6,6 +6,9 @@ import { Table, Td, Th } from '@storybook/components';
 import PropVal from './PropVal';
 import PrettyPropType from './types/PrettyPropType';
 
+const willItOccurAgain = (collection, item, itemIndex = 0) =>
```
Comment (wuweiweiwu) [inline addons/info/src/components/PropTable.js]: Any particular reason that there is a default? Seems like its always specified

File: addons/info/src/components/PropTable.js [hunk 3]
```diff
@@ -6,6 +6,9 @@ import { Table, Td, Th } from '@storybook/components';
...
+  collection.indexOf(item, itemIndex + 1) !== -1;
```
Comment (wuweiweiwu) [inline addons/info/src/components/PropTable.js]: Also a bit hesitant to use this to generate a key, seems a bit unperformant and a bit of an overkill

**Review by ndelangen — APPROVED** [review]

File: addons/info/src/components/PropTable.js (hunk 3 above)
Comment (wuweiweiwu) [reply addons/info/src/components/PropTable.js]: I think that would be good! It'll simplify the logic which is good

**Review by Keraito — APPROVED** [review]

**Pull request merged**